From 0130c1d0d5ffa061b458cbd3e6a817fb82c2ecaf Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 25 May 2022 12:57:36 +0300
Subject: [PATCH] [SW-0] gaudi3: replace address and mask with macro

The register for d2d link initialization are updated
in specs, so replace register address and mask with macros.

Change-Id: I05be6709ffb590745c4d42ce096cd6ab45821dc7
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index a4ff4e1..9cae10e 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -245,27 +245,23 @@
 
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_D2D_LINK_START);
 
-	/*
-	 * TODO: replace with register name and masks
-	 * currently not defined in specs git
-	 */
 	/* Check BUSY bit for 0 value, expected to be 0 */
-	val = regmap_readl(0xC48B0FC);
-	if (val & 0x10) {
+	val = regmap_readl(mmD0_PSOC_GLOBAL_CONF_DPHY_FSM_STS);
+	if (val & D0_PSOC_GLOBAL_CONF_DPHY_FSM_STS_BUSY_M) {
 		arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY);
 		return;
 	}
 
 	/* Set START bit for D2D link initialization */
-	regmap_writel(BIT(0), 0xC48B0F8);
+	regmap_writel(D0_PSOC_GLOBAL_CONF_DPHY_FSM_CTRL_START_M, mmD0_PSOC_GLOBAL_CONF_DPHY_FSM_CTRL);
 
 	/*
 	 * Poll 10ms for BUSY bit to be 0 indicating link initialization
 	 * is completed successfully.
 	 */
 	while (1) {
-		val = regmap_readl(0xC48B0FC);
-		if (!(val & 0x10))
+		val = regmap_readl(mmD0_PSOC_GLOBAL_CONF_DPHY_FSM_STS);
+		if (!(val & D0_PSOC_GLOBAL_CONF_DPHY_FSM_STS_BUSY_M))
 			break;
 
 		retry--;
